rts/lra/lra-coordinator-jar/src/main/java/io/narayana/lra/coordinator/domain/model/LRARecord.java

```diff
@@ -508,7 +507,7 @@ private int retryGetEndStatus(URI endPath, boolean compensate) {
                         return TwoPhaseOutcome.HEURISTIC_HAZARD;
                     case FailedToClose:
                     case FailedToCancel:
-                        return TwoPhaseOutcome.FINISH_ERROR;
+                        return reportFailure(compensate, endPath, "unknown");
```

Comment:
@mmusgrov this will be used in the log message on line https://github.com/jbosstm/narayana/pull/1603/files#diff-0f226ae6945188e648e485306754f30cR473. So "LRARecord: participant %s reported a failure to %s (cause unknown)" but we know that it is already in failed state, so I would rather say (cause already in failed state) but it's not that important.

Reply:
Let's revisit these reasons when we have better reporting of the reasons why participants fail to finish cleanly.